### DISCUSSION TITLE
Mysql connection econnrefused

### DISCUSSION THREAD

**Discussion by EduZeroum** [Budibase/budibase#1981] [Help]
I can't connect my localhost mysql to budibase it says econnrefused , i downloaded mysql server and create tables, the ping works fine but budibase can't connect to mysql, any ideas? I will post some prints
![image](https://user-images.githubusercontent.com/87012689/125429476-2d8570fa-995e-438d-9423-292e168475dc.png)
![image](https://user-images.githubusercontent.com/87012689/125429504-de2f80d5-fe3d-4744-aa23-b99abb89c3b4.png)

**Comment by shogunpurple** [comment]
Hi @EduZeroum - this is possibly the same problem here: https://github.com/Budibase/budibase/discussions/1839

The docker container is trying to reach `127.0.0.1` in it's own network, you may need to use `host.docker.internal`.

**Reply by EduZeroum** [reply]
Thank you, it worked! But now there is another error... I have the latest mysql client. 
![image](https://user-images.githubusercontent.com/87012689/125430665-19137431-0a44-44ed-9674-008e17104c1a.png)

**Reply by shogunpurple** [reply]
Looks like this one: https://github.com/Budibase/budibase/issues/1899

We need to upgrade our MySQL client to work with the latest MySQL server. 

In the meantime, if this is just a personal project and you want to get it working, have a look at the following:

https://stackoverflow.com/questions/50093144/mysql-8-0-client-does-not-support-authentication-protocol-requested-by-server
